tool-openssl/dgst.cc

```diff
@@ -16,14 +16,13 @@ static const argument_t kArguments[] = {
      "Create a hashed MAC with the corresponding key"},
     {"", kOptionalArgument, ""}};
 
-static bool dgst_file_op(const std::string &filename, const EVP_MD *digest,
-                         const char *hmac_key, const size_t hmac_key_len) {
-  ScopedFD scoped_fd = OpenFD(filename.c_str(), O_RDONLY | O_BINARY);
-  int fd = scoped_fd.get();
-
+static bool dgst_file_op(const std::string &filename, const int fd,
```

Comment:
This feels like two functions crammed into one with almost no common code between them (only the first two lines?).  I would have separate `hmac_file_op` and `dgst_file_op` functions.

Reply:
Great point! I had been fixated on finding a way to reuse the operations for `dgstTool` and `md5Tool`..
OpenSSL had muddled these together in a similar pattern here and I followed that, will change to make more sense.